### ISSUE TITLE
Follow posts to be alerted of new comments

### ISSUE THREAD

**Issue by bjoern-tantau** [LemmyNet/lemmy#3069]
- [x] Did you check to see if this issue already exists?
- [x] Is this only a single feature request? Do not put multiple feature requests in one issue.
- [ ] Is this a question or discussion? Don't use this, use https://lemmy.ml/c/lemmy_support.
- [ ] Is this a UI / front end issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo.

**Describe the feature request below**

It would be awesome if we could follow a post to be alerted of new comments added.

This is especially useful for posts asking for stories like https://lemmy.world/post/93996

**Comment by elishamint** [comment]
As we are at it, why stop with posts? I'd suggest also having such alerts with comment sub-trees would be nice.

**Comment by erlend-sh** [comment]
Following #3562 - might this be doable as a plug-in?

**Comment by Nutomic** [comment]
@erlend-sh Maybe. Plugins can get notified about all new posts and comments so that part is fine. Right now there is no way to store which user followed which object, so some storage option would have to be added. But most importantly there would have to be some UI support for subscribing from lemmy-ui and apps. With that in mind it probably makes more sense to implement this in Lemmy directly.  Honestly the whole notification system is very basic and could use a rework to be more flexible.

**Comment by dessalines** [comment]
Yep, notifiations are important enough to be built directly into lemmy. I have some preliminary things I'd like to finish like #2444 that I need to get done before doing a notifications rework.
